sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -485,7 +485,65 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
    * @since 1.4.0
    */
   def saveAsTable(tableName: String): Unit = {
-    saveAsTable(df.sparkSession.sessionState.sqlParser.parseTableIdentifier(tableName))
+    import df.sparkSession.sessionState.analyzer.{AsTableIdentifier, CatalogObjectIdentifier}
+    import org.apache.spark.sql.catalog.v2.CatalogV2Implicits._
+
+    val session = df.sparkSession
```

Comment:
Nit: the local variable is used only once.

Reply:
I'll use it in the follow up :)